### DISCUSSION TITLE
enhancement in webgui: show tTarget and tValue in status

### DISCUSSION THREAD

**Discussion by harvey637** [Lenti84/VdMot_Controller#79] [Ideas]
Hi,
to get a better insight on the working vdmot_controller it might be a nice enhancment to show the tValue and tTarget together with each  valve in the status tab in added columns.
This makes it easy to see, wether a connection to a temperatur provider is working and to see the result in the target and the % columns.

just my idea :-)
cu
Harvey

**Comment by harvey637** [comment]


**Reply by harvey637** [reply]
You are right, doing something fast with floor heating is not "very fast". At least if I have my bathroom window open (for some hours or possibly forgotten) I dont want to spoil more energie in that room. May be it is not worth too much work, just keep in mind :-)
My windows contact is coupled with the thermostat. this is done by scripts (not directly coupled) to do something as "wait for 1 minute" for the open window and "keep temperature down" for 15 more minutes. Not really usefull (in terms of instantenious action by the floor heating) but with the intention of energy saving :-)

With cooling:
I have Bosch heatpump with inverting for cooling in the summer (in extreme situations). So I get the information about actually coling from my heatpump. Homematic thermostats have heating/cooling and programs with inverted logic. The original falmot-c12 also has some logik. If free hours come to plan cooling in vdmot_controller pls ask, i have some graphs and scripts from last summer and around 5 days with cooling. This together with PV, so cooling doesent cost much, as you want cooling when there is plenty of pv energy.
cu
Harvey

**Comment by harvey637** [comment]
now cooling pics:
![grafik](https://user-images.githubusercontent.com/10156426/224294875-4c03bb21-ed79-40bd-9194-dd5c7eeee0a0.png)

You can see:
cold nights, so a little bit of heating (very small orange bars in the night, green line is outside temperature)
hot afternoons, so switch into cooling (blue bars)
the red line is HP poweroutput (it is always given positive, but you could argue that cooling is negative power :-)
The values in the legend are actual values, not from the time of the screenshot.
the temperature is the one of the HP-thermostat (integral part of the HP-system, eg. for inhibiting cooling below dew point temperature)

![grafik](https://user-images.githubusercontent.com/10156426/224297660-263f4d45-91ea-4e4b-8691-31df3db60ace.png)
You can see, that during the small heating peeks the valves are not opened as room temperature is above 
with the rising heat coling starts, the valves are forced into full open (by falmot-c12 signaled by switching to cooling_mode via script)
the rise of the room temperature is less than without cooling
the second day the cooling (initated by HP) starts very late, now really usefull :-(
the open door is not taken into account during cooling period (still working on this)
before the opening of the door has LOWERed the target temperature
inside cooling the target temperatur is HIGHER (coming from the thermostat with the cooling_profile nr4), no open door adjustment
changes in target temperature inside cooling come from the profile nr 4 of thermostat, after cooling going vback into profile nr 1, which is shortly interruppted by door opening (late 24.06.) 
directly after door closing the profile nr 1 changes the temperature again (as always, nothing to do with cooling)

quite complicated to talk over some screenshots, but if you have any questions please ask!
cu
Harvey

**Reply by harvey637** [reply]
Thank you for the reply!

after some testing the need for a "rapid" movement is not so important anymore. Why: I opened the windows of a "long running" valve aand it "very fast" jumps to a much lower percentage. Possibly because of a "learned" PID/AI or because the temperature jump is quite large.
![grafik](https://user-images.githubusercontent.com/10156426/224316908-f1e6fb41-cf7e-4ede-930f-0351ad58a6a2.png)

Although I have no exact idea, WHEN (after what time or with the next (unpredictalbe) adjustment from the PID/AI done) a manually overridden 0% comes back, and it comes back to the percentage it would have been without manuall override interruption.
![grafik](https://user-images.githubusercontent.com/10156426/224317144-6c6a813a-676c-4ed1-be7e-676c8fce3296.png)

Now I finished (more or less) the migration, lets hope for a big smile from my family!
I will wait for some real life windows openings and record the results. And will try to sell my falmot-c12 :-)
And watchout for some homeoffice/office days and some weekends and ... with lots of tTarget changes.
Hopefully few reboots due to updates or config changes ....

have a nice friday!
cu
Harvey

**Reply by SurfGargano** [reply]
The great jump comes from the proportional part of the pi controller depending of the xp parameter
Just pushed the beta 2 with small web-ui fixes

**Reply by harvey637** [reply]
ok, installed Beta2.

Configured all valves to warking from 5%-100% - save - reboot - calibration and so on ....

First observations:
in the status window some valves randomly go from "target:53" (or other values) to "target:0", shortly after state goes to "closes" until "active:0" is reached, 
Than randomly goes on and off forever ... not so good.
![grafik](https://user-images.githubusercontent.com/10156426/224355361-43f0f5ab-7d37-4c0f-bec0-175d8b829576.png)

15:30 update and some reboots
cu
Harvey

**Reply by harvey637** [reply]
Hi,
no, I dont set %valves anymore, only the tValue and TTarget, comming from real sensors, so they are not jumping.
%valves was just testing with windows contacts, I was happy with the behaviour of vdmot_controller so I canceld that.
Only sending tValue and tTarget, and only when there is a change for an individual valve (% to be calculated from vdmot).

So my mqtt is not sending anything now, only reading values and storing "actual" into database for graphing.

After setting active zone to 0/100 and rebooted now the same thing is happening, It looks like two valves are "instable", but NOW after reboot two OTHER valves are opening/closing all the time.

Serial adapter in the place wher vdmot controller is now is complicated, I have to ask someone for a notebook.
Syslog I will try to install one ....

I was so happy before :-((((((
cu
Harvey

**Reply by harvey637** [reply]
ok syslog is up and receiving, mostly onewire stuff, but valves are not yet idle nor in correct % nor making nonsense ...

Anything special you want to be filltered?
cu
Harvey

**Comment by harvey637** [comment]


**Reply by SurfGargano** [reply]
Last idea for today : it is possible that the brokerport you use is busy by another adapter or something ? Port 1883 is default with iobroker

**Reply by SurfGargano** [reply]
I have uploaded the beta3 : 
Changing the control parameter doesn't resets the system anymore.
Added cooling function.

- Set to cooling/heating via webui-control , mqtt or json
- webui : control configuration does not reset the system anymore

set heating/cooling via json : 
http://yourIP/valvesctrlconfig
{"common":{"heatControl":1"}}

0 : Manual
1: on / Heating (the logik where tValue < tTarget opens the valve)
2: on / Cooling (the logik where tValue > tTarget opens the valve)
3: off/Park position

to save permanent : 

http://yourIP/cmd
{action:"vCtrlSave"}

**Comment by harvey637** [comment]
I dont understand the options to {"common":{"heatControl":1"}}
1= heating (the logik where tValue < tTarget opens the valve)
2= cooling (the logik where tValue > tTarget opens the valve)
but what is 0=heating/cooling? Or is this "Off" with manual %values transported into the valves?

Do I understand correctly that changes to "control" are simply saved and immedeatly used and no reboot happens (thus eliminating the long time through calibration and so)?

**Reply by SurfGargano** [reply]
Sorry, the description of heating was my bad.
0 : Manual
1: on / Heating (the logik where tValue < tTarget opens the valve)
2: on / Cooling  (the logik where tValue > tTarget opens the valve)
3: off/Park position

If heating control was set to heat and the valve was not allowed to heat, then the valve goes to the lower 'active zone'
(And also the  'linked' valves)
Also this is for heating control set to cool and the valve is not allowed to cool.

Changes to control are simply saved without reboot.

for your bathroom : you can playing with the parameter ;
xp gives you a immediately step (proportional part)
Ti controls more rapidly, but it may step into a great swing over or under
Ts is the sampling time, after that a new valve position is calculated


The software of the stm was unchanged, so it is not needed to update

**Comment by harvey637** [comment]
ok I am on the way with b3 the second ... wainting for the first valve movements and calibration ...
